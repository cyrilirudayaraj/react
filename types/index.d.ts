import { FormikProps } from 'formik';

interface BaseType {
  id: string;
  name: string;
  description: string;
}
export interface NewTask {
  name: string;
  priorityId: string;
  priorityReasonId: string;
  priorityReasonNote: string;
  originatingSystemId: string;
  originatingCaseId: string;
  taskTypeId: string;
  legacyRuleId: string;
  legacyTaskId: string;
  businessRequirementName: string;
  businessRequirementId: string;
  dueDate: string;
  assignedTo: string;
  businessRequirementValue: string;
  departmentOriginId: string;
  brUpdateReasonIds: any[];
  clientDueDate?: string;
  isArchivedRule?: boolean;
  isLegacyRuleExist?: boolean;
}

export interface TaskStep {
  workflowStepId: any;
  signOffStatement: any;
  signedOffYn: any;
  name: string;
  ordering: string;
  transitionText: any;
  id: string;
  assignedTo: string | null;
  version: string;
  teamId?: string;
  plannedCompletion: string | null;
  actualCompletion: string | null;
  totalWorkLog?: string;
}

export interface TaskDTDetails {
  id: string;
  taskId: string;
  modelName: string;
  name: string;
  refId: string;
  description: string;
  refUrl: string;
  ordering?: string;
  lastModified?: string;
  lastModifiedBy?: string;
  deleted?: string;
  deletedBy?: string;
  version: string;
  expanded?: boolean;
}

export interface TaskDetail {
  originatingSystemId: string | '';
  statusId: string;
  dueDate: any;
  status: string;
  createdBy: string;
  contextId: string;
  contextName: string;
  contextRefId: string;
  ruleReportingCategoryId: any;
  businessRequirementId: string;
  originatingCaseId: string;
  clientDueDate: any;
  modelDesignChanges: any;
  localRuleUseCaseId: any;
  id: string;
  version: string;
  activeTaskStepId: string;
  activeWorkflowStepId: string;
  businessRequirementTypeId: any;
  priorityName: string;
  dtChangesYn: any;
  legacyRuleId: string;
  jiraId: string | null;
  legacyTaskId: string | null;
  legacyRuleArchived: string | null;
  businessRequirementName: string;
  name: string;
  taskTypeId: string;
  taskTypeName: string;
  priorityId: string;
  description: any;
  taskSteps: TaskStep[];
  created: string;
  lastModified: string;
  businessRequirementDesc: any;
  athenaNetChangesYn: any;
  lastModifiedBy: string;
  deploymentDate: string;
  version: string;
  testClaimExample: any;
  testNote: any;
  modelBranchUrl: string;
  dtBranchUrl: string;
  taskDecisionTableDetails: TaskDTDetails[];
  taskDependencies: TaskDependencyDetails[];
  originatingSystemName: string;
  escalatedYn: string;
  associatedBRTasks: TaskInfo[];
  canSignOff?: any;
  ruleEngineId: any;
  ruleTypeId: any;
  visitRuleDisplayLocationId: any;
  internalFixText1: string;
  internalFixText2: string;
  rules2TransformationStatusId: number;
  departmentOriginId: number;
  departmentOriginName: string;
  brUpdateReasons: any[];
  brUpdateReasonIds: any[];
  claimRuleCategoryId: string | null;
  scrubTypeId: string;
  claimRuleCategoryRefId: string;
  claimRuleCategory: string;
  rtSynchronizationLog: any;
  priorityReasonId: any;
  priorityReason: string;
  priorityReasonNote: any;
  rejectionReason: string;
  rejectionDescription: string;
}

export interface DependencyDetails {
  dependencyId: string;
  dependencyName: string;
  dependencySystemId: string;
  deploymentDate: string;
  statusName: string;
  activeWorkflowStepId?: string;
}

export interface TaskDependencyDetails {
  dependencyDate: string;
  dependencyId: string;
  dependencyName: string;
  dependencySystemId: string;
  dependencySystemName: string;
  deploymentDate: string;
  id: string;
  notes: string;
  ordering: string;
  statusName: string;
  taskId: string;
  dependencyCondition: string;
  version: string;
  dependencySystemId: string;
  completedYn: string;
  description: string;
  deleted?: string;
}
export interface TaskInfo {
  id: string;
  name: string;
  activeTaskStepId: string;
  activeTaskStepName: string;
}
export interface TaskType {
  id: string;
  name: string;
  description: string;
}

export interface ReportType {
  id: string;
  name: string;
  description: string;
}

export interface Task {
  activeTaskStepId: string;
  assignedTo: string | null;
  businessRequirementId: string;
  dueOn: string;
  lastModified: string;
  legacyRuleId: string;
  priorityName: string;
  statusName: string;
  taskId: string;
  taskName: string;
  taskTypeId: string;
  taskTypeName: string;
  workflowStepName: string;
  escalatedYn: string;
}

export interface UpdateTaskProps {
  task: TaskDetail;
  taskUpdateAction?: fn;
  toggleEdit?: fn;
  editBtnClick?: any;
  editBtnValue?: any;
  getEditBtnValue?: any;
  leftSectionActive?: any;
  rightSectionActive?: any;
  setLeftFormAction?: fn;
  setRightFormAction?: fn;
}

export type ToastAlertTypes = 'attention' | 'info' | 'success';

export interface ToastMessage {
  id?: string;
  headerText: string | fn;
  alertType?: ToastAlertTypes;
  message: string | fn;
  params?: Record<string, unknown>;
}

export interface ReportCriteria {
  reportTypeId: string;
  dateType: string;
  startDate: date;
  endDate: date;
}

export interface UserComment {
  id: string;
  taskId: string;
  content: any;
  created: string;
  createdBy: string;
  createdTimestamp: string;
}

export interface TaskDiscusisonProps {
  task: TaskDetail;
  fetchUserCommentDetails?: fn;
  taskId: any;
  commentId: any;
  resetCommentId: any;
}

export interface DiscussionFormProps {
  taskdetails: TaskDetail;
  createUserComment: fn;
  comments: UserComment[];
  resetPopupActions?: any;
  resetCommentId: any;
}

export interface UserCommentFormProps {
  formik: FormikProps<any>;
}

export type BusinessRequirementType = BaseType;

export interface RuleReportingCategory extends BaseType {
  refId: string;
}

export interface Context extends BaseType {
  refId: string;
}

export type LocalRuleUseCase = BaseType;

export interface AttachmentDetail {
  id: string;
  taskId: string;
  taskStepId: any;
  description: string;
  attachmentTypeId: string;
  attachmentName: string;
  createdBy: string;
  fileName: string;
  filePath: string;
  version: string;
}

export interface TaskDocumentProps {
  task: TaskDetail;
  fetchAttachmentDetails?: fn;
}

export interface AttachmentType {
  id: string;
  name: string;
  version: string;
}
export type WorkFlowFilterType = BaseType;

export interface StatusFilterType {
  workFlowStepQueueIds: string;
  name: string;
  id: string;
}
export interface TimeEntry {
  id: string;
  taskId: string;
  taskStepId: any;
  spentOn: string;
  spentTimeInMins: string;
  workflowStepName: string;
  workflowStepId: string;
  createdBy: string;
  created: string;
  version: string;
  totalSpentTimeInMins: string;
}
export interface WorkLogProps {
  task: TaskDetail;
  fetchTimeEntryLogs?: fn;
}

export interface AuditEntry {
  fieldName?: string;
  oldValue?: Record<string, string | null> | string | null;
  newValue?: Record<string, string | null> | string | null;
}

export interface AuditEvent {
  eventType: string;
  objectType: string;
  eventName: string;
  created: string;
  createdBy: string;
  changes: AuditEntry[];
}

export interface Shape {
  text: string;
  value: string;
}

export interface Deployment {
  branchName: string;
  created: string;
  createdBy: string;
  dcUrl: string;
  decisionServiceName: string;
  deploymentConfigName: string;
  description: string | null;
  id: any;
  name: string;
  releaseVersion: string | null;
  status: string;
  statusId: string;
  verifiedYn: string | null;
  version: any;
  archiveFileName: string;
  archiveFilePath: string;
  envNames?: string;
  taskIds?: string;
}

export interface DeploymentLog {
  action: string;
  category: string | null;
  id: string;
  message: string;
  name: string;
  ordering: string;
  request: string;
  response: string;
  statusId: string;
  version: string;
}

export interface DeploymentEnv {
  id: string;
  initialVersion: string;
  name: string;
  resUrl: string;
  resultVersion: string;
  statusId: string;
  version: string;
}

export interface DeploymentTask {
  businessRequirementId: string;
  businessRequirementName: string;
  id: string;
  legacyRuleId: string;
  name: string;
  taskTypeId: string;
  taskTypeName: string;
}

export interface DeploymentDetails extends Deployment {
  logs: DeploymentLog[];
  tasks: DeploymentTask[];
  envs: DeploymentEnv[];
}

export interface DepartmentOrigin {
  id: string;
  name: string;
  description: string;
}

export interface BRUpdateReason {
  id: string;
  name: string;
  description: string;
}

export interface DeploymentStatus {
  ordering: string;
  name: string;
  id: string;
}

export interface PriorityPolicy {
  description1: string;
  description2: string;
  externalDueDate: string;
  externalTatInDays: string;
  id: string;
  internalDueDate: string;
  internalTatInDays: string;
  name: string;
}

export interface RejectionReason {
  id: string;
  name: string;
  version: string;
  ordering: string;
}
export interface RulesMigrationDetail {
  id: string;
  ruleId: string;
  contextId: string;
  businessRequirementId: string;
  migrationStatus: string;
  comment: string;
}

export interface BRDetails {
  legacyRuleId: string;
  rules2TransformationStatusId: string;
  rules2TransformationStatus: string;
  name: string;
  ruleReportingCategoryId: string;
  ruleReportingCategoryName: string;
  description: string;
  internalFixText2: string;
  ruleTypeId: string;
  ruleType: string;
  internalFixText1: string;
  localRuleUseCaseId: string;
  localRuleUseCaseName: string;
  contextId: string;
  contextName: string;
  id: string;
  visitRuleDisplayLocationId: string;
  visitRuleDisplayLocation: string;
  businessRequirementTypeId: string;
  businessRequirementType: string;
  associatedBRTasks: any[];
}
