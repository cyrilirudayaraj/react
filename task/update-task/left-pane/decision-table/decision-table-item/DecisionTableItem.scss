@import '~@athena/forge/sass/forge-abstracts';
.task-dt {
  .delete-icon {
    .fe_c_icon__icon--primary {
      fill: #d32e45;
    }
  }

  .dt-url {
    padding-top: map-get($global-padding, 'large');
  }

  .dt-url-text {
    padding-top: 5px;
    padding-left: 5px;
  }

  .dt-url a {
    display: inline-flex;
  }

  .dt-url a:hover,
  .dt-url a:visited,
  .dt-url a:focus,
  .dt-url a:link {
    box-shadow: none;
    color: none;
    border: none;
    transition: none;
    outline: none;
  }

  .padding-bottom {
    padding-bottom: 0 !important;
  }

  .max-width .fe_l_grid-row__col {
    min-width: 100%;
  }

  .task-dt .fe_c_form-field {
    padding-bottom: 0 !important;
  }
}
.hide {
  display: none !important;
}
