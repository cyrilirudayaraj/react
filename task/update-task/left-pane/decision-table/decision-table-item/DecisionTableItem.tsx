import React, { useState, useEffect } from 'react';
import {
  Accordion,
  AccordionItem,
  Icon,
  GridRow,
  GridCol,
  FormField,
  Button,
} from '@athena/forge';
import { FormikProps } from 'formik';
import './DecisionTableItem.scss';
import Labels from '../../../../../../constants/Labels';
import { get, isArray } from 'lodash';
import StringUtil from '../../../../../../utils/StringUtil';
import WFTextarea from '../../../../../../components/wf-textarea/WFTextarea';
import { isTaskEditable } from '../../../UpdateTask';
import { TaskDetail } from '../../../../../../types';
import Acl from '../../../../../../constants/Acl';
import WFEnableForPermission from '../../../../../../components/wf-enableforpermission/WFEnableForPermission';
import RichTextEditor from '../../../../../../components/rich-text-editor/RichTextEditor';

interface TaskDecisionTableItemProps {
  index: number;
  formik: FormikProps<any>;
  deleteAction: (event: any) => void;
  deleted: boolean;
  mandatoryFields?: any;
  expanded: boolean;
  task?: TaskDetail;
}

const getFormFieldProps = (fieldName: string, formik: any) => {
  const props = {
    ...formik.getFieldProps(fieldName),
    required: true,
    labelWidth: 1,
    autoComplete: 'off',
    error: get(formik.errors, fieldName),
    id: fieldName,
  };
  return props;
};

export default function DecisionTableItem(
  props: TaskDecisionTableItemProps
): JSX.Element {
  const labels = Labels.TASK_DT_DETAILS;
  const [expanded, setExpanded] = useState(props.expanded);
  useEffect(() => {
    setExpanded(props.expanded);
  }, [props.expanded]);
  const { index, formik, deleteAction } = props;
  const isDisabled = !isTaskEditable(props.task);

  const getDeleteDTTemplate = (): JSX.Element => {
    return (
      <div className="delete_dt">
        <WFEnableForPermission
          permission={Acl.TASK_DECISION_TABLE_CHANGES_UPDATE}
        >
          <Button
            className="delete-icon"
            onClick={(event) => {
              if (!isDisabled) deleteAction(event);
            }}
            icon={Labels.TASK_DT_DETAILS.DELETE}
            variant="tertiary"
            disabled={isDisabled}
          />
        </WFEnableForPermission>
      </div>
    );
  };

  const getHeaderText = (): string => {
    let headerText = formik.getFieldProps(
      `taskDecisionTableDetails.${index}.refId`
    ).value;
    if (headerText === 0) {
      headerText = headerText.toString();
    }
    return headerText ? StringUtil.formatDecisionTableId(headerText) : '';
  };

  const getModelNameTemplate = (): JSX.Element => {
    return (
      <GridCol width={{ small: 8, medium: 8, large: 8 }}>
        <WFEnableForPermission
          permission={Acl.TASK_DECISION_TABLE_CHANGES_UPDATE}
        >
          <WFTextarea
            {...props}
            label={labels.DT_MODEL_NAME}
            labelWidth={12}
            name={`taskDecisionTableDetails.${index}.modelName`}
            maxlength={200}
            className="max-width padding-bottom"
            disabled={isDisabled}
          />
        </WFEnableForPermission>
      </GridCol>
    );
  };

  const getDesisionTableNameTemplate = (): JSX.Element => {
    return (
      <GridCol width={{ small: 6, medium: 6, large: 6 }}>
        <WFEnableForPermission
          permission={Acl.TASK_DECISION_TABLE_CHANGES_UPDATE}
        >
          <WFTextarea
            {...props}
            label={labels.DT_NAME}
            labelWidth={12}
            name={`taskDecisionTableDetails.${index}.name`}
            maxlength={400}
            className="max-width padding-bottom"
            disabled={isDisabled}
          />
        </WFEnableForPermission>
      </GridCol>
    );
  };

  const getDecisionTableRefIdTemplate = (): JSX.Element => {
    return (
      <GridCol width={{ small: 2, medium: 2, large: 2 }}>
        <WFEnableForPermission
          permission={Acl.TASK_DECISION_TABLE_CHANGES_UPDATE}
        >
          <FormField
            labelText={labels.DT_ID}
            labelWidth={12}
            type="number"
            max="999999"
            {...getFormFieldProps(
              `taskDecisionTableDetails.${index}.refId`,
              formik
            )}
            className="max-width padding-bottom"
            disabled={isDisabled}
          />
        </WFEnableForPermission>
      </GridCol>
    );
  };

  const getUrlLinkTemplate = (): JSX.Element => {
    let url = formik.getFieldProps(`taskDecisionTableDetails.${index}.refUrl`)
      .value;
    url = StringUtil.getValidUrl(url);
    return (
      <GridCol width={{ small: 2, medium: 2, large: 2 }}>
        <div className="dt-url">
          <a href={url} target="_blank" rel="noopener noreferrer">
            <div>
              <Icon icon="NavigateAway" />
            </div>
            <div className="dt-url-text">
              <span>{Labels.TASK_DT_DETAILS.GO}</span>
            </div>
          </a>
        </div>
      </GridCol>
    );
  };

  const getClassName = (): string => {
    let className = 'task-dt';
    if (props.deleted) {
      className = 'hide';
    }
    return className;
  };

  const getUrlTemplate = (): JSX.Element => {
    return (
      <GridCol width={{ small: 10, medium: 10, large: 10 }}>
        <WFEnableForPermission
          permission={Acl.TASK_DECISION_TABLE_CHANGES_UPDATE}
        >
          <FormField
            labelText={labels.DT_Link}
            labelWidth={12}
            {...getFormFieldProps(
              `taskDecisionTableDetails.${index}.refUrl`,
              formik
            )}
            className="ref-url padding-bottom"
            disabled={isDisabled}
          />
        </WFEnableForPermission>
      </GridCol>
    );
  };

  const getTaskDTDescriptionTemplate = (): JSX.Element => {
    const descriptionId = `taskDecisionTableDetails.${index}.description`;
    return (
      <GridCol width={{ small: 11, medium: 11, large: 11 }}>
        <WFEnableForPermission
          permission={Acl.TASK_DECISION_TABLE_CHANGES_UPDATE}
        >
          <RichTextEditor
            id={descriptionId}
            // Starting wrapper id from 30+ because other editor fields will start from 1.
            // Just to avoid conflict added 30 to index
            wrapperId={index + 30}
            labelText={labels.DT_CHANGE_DESCRIPTION}
            formik={formik}
            isDisabled={isDisabled}
            maxLength={2000}
            isRequired={props?.mandatoryFields[descriptionId] || false}
          ></RichTextEditor>
        </WFEnableForPermission>
      </GridCol>
    );
  };

  const handleAccordionClick = (event: any) => {
    if (
      event.target.className !== 'fe_c_accordion-item__header' &&
      event.target.className !== 'fe_c_accordion-item__header fe_is-expanded' &&
      event.target.className !==
        'fe_c_heading fe_c_heading--section fe_c_accordion-item__heading-text'
    ) {
      return;
    }
    setExpanded(!expanded);
  };

  const error = isArray(formik.errors?.taskDecisionTableDetails)
    ? formik.errors?.taskDecisionTableDetails[index]
    : undefined;
  if (error && !expanded && formik.isSubmitting) {
    setExpanded(true);
  }

  return (
    <Accordion className={getClassName()}>
      <AccordionItem
        onClick={handleAccordionClick}
        headingText={getHeaderText()}
        headerSlot={getDeleteDTTemplate()}
        expanded={expanded}
      >
        <GridRow>{getModelNameTemplate()}</GridRow>
        <GridRow>
          {getDesisionTableNameTemplate()}
          {getDecisionTableRefIdTemplate()}
        </GridRow>
        <GridRow>{getTaskDTDescriptionTemplate()}</GridRow>
        <GridRow>
          {getUrlTemplate()}
          {getUrlLinkTemplate()}
        </GridRow>
      </AccordionItem>
    </Accordion>
  );
}
