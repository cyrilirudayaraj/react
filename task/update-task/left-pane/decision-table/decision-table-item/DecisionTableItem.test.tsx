import { configure, mount } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import DecisionTableItem from './DecisionTableItem';
import { AccordionItem } from '@athena/forge';
import { FormikValues } from 'formik';
import { Provider } from 'react-redux';
import { TaskDetail } from '../../../../../../types';
import configureStore from 'redux-mock-store';

configure({ adapter: new Adapter() });

const taskDetails = {} as TaskDetail;
taskDetails.activeTaskStepId = '1';

const store = configureStore([])({
  task: {
    taskDetails: taskDetails,
  },
  userPermission: {
    userPermissions: ['task.update'],
  },
  updateTaskDetails: jest.fn(),
});
store.dispatch = jest.fn();
describe('DecisionTableItem', () => {
  const props: FormikValues = {
    values: {
      index: 0,
      formik: {},
    },
    touched: {},
    errors: {},
    handleChange: (a) => a,
    handleBlur: (a) => a,
    handleSubmit: (a) => a,
    getFormFieldProps: jest.fn((fieldName: string, formik: any) => {
      return { value: 'test' };
    }),
    getFieldProps: jest.fn(() => {
      return 'tets';
    }),
    deleteAction: (a) => a,
  };

  const wrapper = mount(
    <Provider store={store}>
      <DecisionTableItem formik={props} mandatoryFields={jest.fn()} />
    </Provider>
  );
  it('should render as expected', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('AccordionItem renders properly', () => {
    expect(wrapper.find(AccordionItem)).toHaveLength(1);
  });
});
