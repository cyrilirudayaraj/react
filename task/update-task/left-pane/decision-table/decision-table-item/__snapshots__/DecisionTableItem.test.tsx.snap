// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecisionTableItem should render as expected 1`] = `
"<Provider store={{...}}>
  <DecisionTableItem formik={{...}} mandatoryFields={[Function: mockConstructor]}>
    <Accordion className=\\"task-dt\\">
      <List className=\\"fe_c_accordion task-dt\\" padded={false} dividers={false}>
        <ul className=\\"fe_c_list fe_c_accordion task-dt\\">
          <AccordionItem onClick={[Function: handleAccordionClick]} headingText=\\"\\" headerSlot={{...}} expanded={[undefined]} defaultExpanded={false} headingTag=\\"h3\\" mountedWhileHidden={false} padded={true} tabIndex={0}>
            <ListItem className=\\"fe_c_accordion-item\\" onClick={[Function: handleAccordionClick]}>
              <li className=\\"fe_c_list__item fe_c_accordion-item\\" onClick={[Function: handleAccordionClick]}>
                <div className=\\"fe_c_accordion-item__header\\" onClick={[Function: bound toggleExpand]} onKeyDown={[Function: onKeyDown]} aria-expanded={false} role=\\"button\\" tabIndex={0}>
                  <Icon icon=\\"Expand\\" title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\" iconSet={{...}}>
                    <Expand title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\">
                      <IconBase viewBox=\\"0 0 25 25\\" title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\">
                        <svg xmlns=\\"http://www.w3.org/2000/svg\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_icon fe_c_accordion-item__expand\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                          <title>
                            Expand
                          </title>
                          <g>
                            <g className=\\"fe_c_icon__background\\">
                              <rect width=\\"25\\" height=\\"25\\" />
                            </g>
                            <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                              <polygon points=\\"12.5 16.6 6 10 7 9 12.5 14.4 18 9 19 10 12.5 16.6\\" />
                            </g>
                          </g>
                        </svg>
                      </IconBase>
                    </Expand>
                  </Icon>
                  <Heading text=\\"\\" variant=\\"section\\" headingTag=\\"h3\\" className=\\"fe_c_accordion-item__heading-text\\">
                    <h3 className=\\"fe_c_heading fe_c_heading--section fe_c_accordion-item__heading-text\\" />
                  </Heading>
                  <div className=\\"fe_c_accordion-item__header-slot\\">
                    <div className=\\"delete_dt\\">
                      <Connect(WFEnableForPermission) permission=\\"task.decision-table-changes.update\\">
                        <WFEnableForPermission permission=\\"task.decision-table-changes.update\\" userPermissions={{...}} dispatch={[Function: mockConstructor]}>
                          <ForwardRef className=\\"delete-icon\\" onClick={[Function: onClick]} icon=\\"Delete\\" variant=\\"tertiary\\" disabled={false} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                            <button className=\\"fe_c_button fe_c_button--tertiary fe_c_button--medium delete-icon\\" aria-disabled={false} aria-label=\\"Delete\\" onClick={[Function: handleClick]} type=\\"button\\">
                              <div className=\\"fe_c_button__content\\">
                                <Delete title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--tertiary\\">
                                  <IconBase viewBox=\\"0 0 25 25\\" title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--tertiary\\">
                                    <svg xmlns=\\"http://www.w3.org/2000/svg\\" className=\\"fe_c_icon fe_c_button__icon fe_c_button__icon--tertiary\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                                      <g>
                                        <g className=\\"fe_c_icon__background\\">
                                          <rect width=\\"25\\" height=\\"25\\" />
                                        </g>
                                        <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                                          <g>
                                            <path d=\\"M22.2,4.4H16.9V3.8A1.9,1.9,0,0,0,15,2H10A1.9,1.9,0,0,0,8.1,3.8v.6H2.8V5.9H4.4V23H17.2a3.4,3.4,0,0,0,3.4-3.3V5.9h1.6ZM19.1,19.7a1.9,1.9,0,0,1-1.9,1.8H5.9V5.9H19.1ZM9.6,3.8c0-.2.2-.3.4-.3h5c.2,0,.4.1.4.3v.6H9.6Z\\" />
                                            <rect x=\\"7.9\\" y=\\"8.6\\" width=\\"1.5\\" height=\\"10.76\\" />
                                            <rect x=\\"11.8\\" y=\\"8.6\\" width=\\"1.5\\" height=\\"10.76\\" />
                                            <rect x=\\"15.6\\" y=\\"8.6\\" width=\\"1.5\\" height=\\"10.76\\" />
                                          </g>
                                        </g>
                                      </g>
                                    </svg>
                                  </IconBase>
                                </Delete>
                              </div>
                            </button>
                          </ForwardRef>
                        </WFEnableForPermission>
                      </Connect(WFEnableForPermission)>
                    </div>
                  </div>
                </div>
                <div className=\\"fe_c_accordion-item__content fe_is-exited\\">
                  <CSSTransition in={false} classNames=\\"fe_is-expanding\\" timeout={400} onEnter={[Function: bound handleEnter]} onEntering={[Function: handleEntering]} onEntered={[Function: handleEntered]} onExit={[Function: handleExit]} onExiting={[Function: handleExiting]} onExited={[Function: bound handleExited]} mountOnEnter={true} unmountOnExit={true}>
                    <Transition in={false} timeout={400} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} onExited={[Function (anonymous)]} mountOnEnter={true} unmountOnExit={true} appear={false} enter={true} exit={true} />
                  </CSSTransition>
                </div>
              </li>
            </ListItem>
          </AccordionItem>
        </ul>
      </List>
    </Accordion>
  </DecisionTableItem>
</Provider>"
`;
